src/main/webapp/app/exercises/shared/result/result.utils.ts

```diff
@@ -288,3 +290,66 @@ export const isBuildFailed = (submission?: Submission) => {
 export const isManualResult = (result?: Result) => {
     return result?.assessmentType !== AssessmentType.AUTOMATIC;
 };
+
+/**
+ * Retrieves a list of test cases names contained in a result's feedback list.
+ *
+ * @param results list of results to extract the test case names from
+ * @return list of extracted test case names
+ * @private
+ */
+export function getTestCaseNamesFromResults(results: ResultWithPointsPerGradingCriterion[]): string[] {
+    const testCasesNames: Set<string> = new Set();
+    results.forEach((result) => {
+        if (!result.result.feedbacks) {
+            return [];
+        }
+        result.result.feedbacks.map((f) => {
+            if (f.type && [FeedbackType.AUTOMATIC, FeedbackType.AUTOMATIC_ADAPTED].indexOf(f.type) >= 0) {
+                testCasesNames.add(f.text ?? 'Test ' + result.result.feedbacks?.indexOf(f) + 1);
+            }
+        });
+    });
+    return Array.from(testCasesNames);
+}
+
+/**
+ * Extracts test case results from a given result and returns them.
+ * If no feedback is found in the result an empty array is returned
+ * @param result from which the test case results should be extracted
+ * @param testCaseNames list containing the test names
+ * @param withFeedback if true, the feedback's full text is included in case of failed test case
+ * @private
+ */
+export function getTestCaseResults(result: ResultWithPointsPerGradingCriterion, testCaseNames: string[], withFeedback?: boolean): TestCaseResult[] {
+    const testCaseResults: TestCaseResult[] = [];
+
+    testCaseNames.forEach((testName) => {
+        const feedback = getFeedbackByTestCase(testName, result.result.feedbacks);
+
+        let resultText = 'Skipped';
+        if (feedback?.positive) {
+            resultText = 'Passed';
+        } else if (!feedback?.positive) {
+            resultText = !!withFeedback && feedback?.detailText ? `Failed: "${feedback.detailText}"` : 'Failed';
+        }
```

Comment:
Either `feedback?.positive` or `!feedback?.positive` will always be true, so the 'Skipped' will never used

Reply:
Changed thanks 👍